controller/webhook/server.go

```diff
@@ -0,0 +1,152 @@
+package webhook
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/linkerd/linkerd2/controller/k8s"
+	pkgTls "github.com/linkerd/linkerd2/pkg/tls"
+	log "github.com/sirupsen/logrus"
+	admissionv1beta1 "k8s.io/api/admission/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/yaml"
+)
+
+type handlerFunc func(*k8s.API, *admissionv1beta1.AdmissionRequest) (*admissionv1beta1.AdmissionResponse, error)
+
+// Server describes the https server implementing the webhook
+type Server struct {
+	*http.Server
+	api                 *k8s.API
+	handler             handlerFunc
+	controllerNamespace string
+}
+
+// NewServer returns a new instance of Server
+func NewServer(api *k8s.API, addr, name, controllerNamespace string, rootCA *pkgTls.CA, handler handlerFunc) (*Server, error) {
+	c, err := tlsConfig(rootCA, name, controllerNamespace)
+	if err != nil {
+		return nil, err
+	}
+
+	server := &http.Server{
+		Addr:      addr,
+		TLSConfig: c,
+	}
+
+	s := &Server{server, api, handler, controllerNamespace}
+	s.Handler = http.HandlerFunc(s.serve)
+	return s, nil
+}
+
+// Start starts the https server
+func (s *Server) Start() {
+	log.Infof("listening at %s", s.Server.Addr)
+	if err := s.ListenAndServeTLS("", ""); err != nil {
```

Comment:
is `ListenAndServeTLS` eventually going to take params?

Reply:
Nope. The certs are added in `s`'s setup above in the `NewServer` function.